JCLOUDS-702: JCloud ProfitBricks provider - NIC & Firewall API

devcsrj opened the pull request:
Hi! This PR implements ProfitBricks' API on: 
* NIC - [live tests](https://gist.github.com/devcsrj/28827969262cf56dd56c); [docs](https://www.profitbricks.com/apidoc/1_3/APIDocumentation.html?NICOperations.html)
* Firewall - [live tests](https://gist.github.com/devcsrj/77dc6768441c40baaf67); [docs](https://www.profitbricks.com/apidoc/1_3/APIDocumentation.html?FirewallOperations.html)

Looking forward to the code review to correct stuff I might've missed. :)

/cc @nacx @jasminSPC

nacx commented on profitbricks/src/main/java/org/jclouds/profitbricks/domain/Firewall.java:
```diff
@@ -0,0 +1,367 @@
+
+      protected void checkPortRange() {
+	 checkArgument((portRangeEnd == null && portRangeStart == null)
+		 || (portRangeEnd != null && portRangeStart != null), "Port range must be both present or null");
```
[nit] This can be more concise:
```java
checkArgument(!(portRangeEnd == null ^ portRangeStart == null), "Port range must be both present or null");
```

devcsrj replied:
Oooh clever. That's something I need to remember

nacx commented on profitbricks/src/main/java/org/jclouds/profitbricks/domain/Server.java:
```diff
@@ -244,188 +255,188 @@ public DescribingBuilder self() {
+	    }
 
-         }
+	 }
```
Take care of the re-formatting of this class. Changing from spaces to tabs?

nacx commented on profitbricks/src/main/java/org/jclouds/profitbricks/util/Strings3.java:
```diff
@@ -0,0 +1,43 @@
+      return MAC_ADDR_PATTERN.matcher(in).matches();
+   }
+
+   public static boolean isIpAddress(String in) {
```
Prefer Guava's `InetAddresses.isInetAddress(String)` to this method?

nacx commented:
Thanks! Code looks pretty good. A couple global comments:

* Take a look at all those unrelated files that are included in this PR that are just reformatted (changing from spaces to tabs). can you remove them from this PR?
* Add a unit test for the Strings3 class, and consider renaming it to `MacAddresses` or similar? :)

devcsrj commented:
Hi @nacx ! As usual, thanks for the quick review!

So I addressed the initial issues and reformatted the whole project to use the correct formatting - this removed unrelated classes such as `ImageListResponseHandler`, `GetProvisioningStateResponseHandler`). This also fixed the diff on files like `Server` and `Storage` to only diff lines that are actually changed.

With that said, here ya go. :)

nacx commented on profitbricks/src/main/java/org/jclouds/profitbricks/util/MacAddresses.java:
```diff
@@ -0,0 +1,33 @@
+
+   public static void main(String[] args) {
+      System.out.println(isMacAddress("AA:BB:CC:DD:EE:FF".toLowerCase()));
+   }
```
Remove this method :)

devcsrj replied:
woops. Haha

nacx commented:
Apart from the main method there, code lgtm. Mind squashing when fixing? I'll merge it then.
As usual, nice contribution! Thanks!

nacx commented:
The Firewall class and the NicApiMockTest classes are usign Windows line endings. Mind fixing them?

nacx commented:
Pushed to master as [c78a66c7](https://git1-us-west.apache.org/repos/asf?p=jclouds-labs.git;a=commit;h=c78a66c7). Thanks @devcsrj!
